static/js/stream_create.js

```diff
@@ -182,12 +182,17 @@ function create_stream() {
     const stream_name = $.trim($("#create_stream_name").val());
     const description = $.trim($("#create_stream_description").val());
     const privacy_setting = $('#stream_creation_form input[name=privacy]:checked').val();
-    const is_announcement_only = $('#stream_creation_form input[name=is-announcement-only]').prop('checked');
+    let stream_post_policy = parseInt($('#stream_creation_form input[name=stream-post-policy]').val(), 10);
     const principals = get_principals();
 
     let invite_only;
     let history_public_to_subscribers;
 
+    // To allow non admins to create streams.
+    if (isNaN(stream_post_policy)) {
+        stream_post_policy = 1;
```

Comment:
Use a numerical value here.  I'd also update the comment to explain __why__ we need this -- e.g.

"Because the stream_post_policy field is hidden when non-administrators create streams, we need to set the default value here."

That said, I think it might be better to not pass a `stream_post_policy` at all if we want the default value?  It's a bit awkward with this interface.

What I probably want to do is refactor this whole flow to have `create_stream` be building up a dictionary to send to the server; the interface for `ajaxSubscribeForCreation` right now is terrible.  If you're interested in doing that as a prep commit (i.e. rebased to the start of this branch), it'd be really nice; I'd probably:
* inline `ajaxSubscribeForCreation` into this function first
* Make a `data` object and change a bunch of this code to be populating that data object
* Then in this commit, which just adds the `stream_post_policy` if available.

But I'm OK with doing it as a follow-up cleanup.

Reply:
> Use a numerical value here. 

Sorry, I didn't get this line. Should i make the condition more explicit by checking whether the value doesn't lie in the 3 available post_policy values?

Reply:
Oh, I think I meant using the `.code` thing; but the rest of the comment suggests we remove this default code entirely anyway.

Reply:
> That said, I think it might be better to not pass a stream_post_policy at all if we want the default value?

Actually, I think we require the `stream_post_policy` values as the admins form creation logic passes through this function as well and then we post the stream subscription data.